rustls/src/common_state.rs

```diff
@@ -307,21 +308,32 @@ impl CommonState {
     }
 
     fn send_single_fragment(&mut self, m: OutboundPlainMessage) {
-        match self
-            .record_layer
-            .next_pre_encrypt_action()
-        {
-            record_layer::PreEncryptAction::Nothing => {}
+        match (
+            &m,
+            self.record_layer
+                .next_pre_encrypt_action(),
+        ) {
+            // Alerts are always sendable -- never quashed by a PreEncryptAction.
+            (
+                OutboundPlainMessage {
+                    typ: ContentType::Alert,
+                    ..
+                },
```

Comment:
Since there is not actually any interaction between the matches for `m` and the next pre-encrypt action, I think this would be easier to follow if you handled the alert before even calling `next_pre_encrypt_action()`?